internal/pkg/util/env/create.go

```diff
@@ -103,7 +103,16 @@ func warnDeprecatedEnvUsage(hostEnvs []string) {
 				if key != "" {
 					legacyEnv := LegacySingularityEnvPrefix + key
 					newEnv := ApptainerEnvPrefix + key
-					sylog.Infof("Environment variable %v is set, but %v is preferred", legacyEnv, newEnv)
+					skip := false
+					for _, e := range hostEnvs {
+						if strings.HasPrefix(e, newEnv) {
```

Comment:
Why isn't this just a check for equal?  Wouldn't checking the prefix make it get matches for variables names that aren't the same but start with the same value?